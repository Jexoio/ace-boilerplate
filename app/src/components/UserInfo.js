// @flow
import React, { Component } from 'react';
import Theme from '../providers/Theme';
import styled from 'styled-components';

const UserInfoDiv = styled.div`
  margin: 10px 0;
`;

type UserInfoProps = {
  me: {
    id: string,
    userAccountId: string,
    clientKey: string,
    updatedAt: string
  }
};
class UserInfo extends Component<UserInfoProps> {
  render() {
    const { me } = this.props;
    return (
      <Theme.Consumer>
        {tokens => (
          <UserInfoDiv>
            <h2 style={{ color: tokens.h1.textColor }}>User info</h2>
            <table>
              <tbody>
                <tr>
                  <td>id</td>
                  <td>{me.id}</td>
                </tr>
                <tr>
                  <td>userAccountId</td>
                  <td>{me.userAccountId}</td>
                </tr>
                <tr>
                  <td>clientKey</td>
                  <td>{me.clientKey}</td>
                </tr>
                <tr>
                  <td>updatedAt</td>
                  <td>{me.updatedAt}</td>
                </tr>
              </tbody>
            </table>
          </UserInfoDiv>
        )}
      </Theme.Consumer>
    );
  }
}

export default UserInfo;
