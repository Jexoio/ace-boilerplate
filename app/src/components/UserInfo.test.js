import React from 'react';
import { shallow, wrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import UserInfo from './UserInfo';

const me = {
  id: 'foo',
  userAccountId: 'bar',
  clientKey: 'baz',
  clientKey: 'boo'
};
it('renders without crashing', () => {
  shallow(<UserInfo me={me} />);
});
it('matches the snapshot', () => {
  const tree = renderer.create(<UserInfo me={me} />).toJSON();
  expect(tree).toMatchSnapshot();
});
