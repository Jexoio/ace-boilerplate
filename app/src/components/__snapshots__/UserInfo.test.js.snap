// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="sc-bdVaJa iqUsw"
>
  <h2
    style={
      Object {
        "color": "#172B4D",
      }
    }
  >
    User info
  </h2>
  <table>
    <tbody>
      <tr>
        <td>
          id
        </td>
        <td>
          foo
        </td>
      </tr>
      <tr>
        <td>
          userAccountId
        </td>
        <td>
          bar
        </td>
      </tr>
      <tr>
        <td>
          clientKey
        </td>
        <td>
          boo
        </td>
      </tr>
      <tr>
        <td>
          updatedAt
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
